### ISSUE TITLE
Sigma and Gephi have inverse y axes

### ISSUE THREAD

**Issue by computermacgyver** [oxfordinternetinstitute/gephi-plugins#5]
Invert the y-coordinates so that layout in browser matches Gephi  (This is especially important for geographic layouts)

**Comment by jthrilly** [comment]
Just out of curiosity.....could this explain any issues with touch events? I could never get them to work properly.

**Comment by computermacgyver** [comment]
I don't know. I wouldn't have thought so because sigma is consistent 
within itself. It's just that the origin is in a different location 
which means the graph is rendered upside down as compared to how it is 
rendered in Gephi. However, thinking about it more if the origin is in 
the bottom-left (instead of the typically top-left) then depending how a 
pan event was sent to Sigma it could result in the graph moving the 
wrong way (i.e. up instead of down). It shouldn't cause issues with 
other directors or with zoom, however, I would think.

**Comment by timdcrain** [comment]
I changed this in Sigma here: 3d9e1faebdef4cd2816936d92173bf50c201d228

This puts the origin in the center, not the bottom left, but it does invert the y-axis so it should match Gephi. Also, this leaves the underlying coordinate system unchanged, so pan events should be preserved (although I have no idea if this is true).

**Comment by computermacgyver** [comment]
Thanks, Tim. This looks like a good solution. I'm linking to the pull request for my own reference:
https://github.com/jacomyal/sigma.js/pull/77

**Comment by NoahGordon** [comment]
How do you invert the y-coordinates so the layout in the browser matches Gephi?

**Comment by computermacgyver** [comment]
@NoahGordon I was waiting on upstream, Sigma.js, and ideally wanted to see this fix there to avoid having different versions of sigma.js floating around. I consider this bug low priority as, in general, graph layout is invariant to the exact position of nodes (i.e., a flipped graph is isomorphically the same graph and the flip doesn't affect its presentation usually [geographic layouts being the one counter-example I've seen]).

Nonetheless, I may address this in an interm release while waiting on upstream. Probably the easiest fix is to alter the js/sigma/sigma.parseJson.js file to add the second line as follows:

```
var id=data.nodes[i].id;
data.nodes[i].y=-data.nodes[i].y;
```

**Comment by ahalterman** [comment]
Thanks @computermacgyver ! I'm working with a geo layout and adding that line to sigma.parseJson.js solved the inversion problem right away.

**Comment by ellenprobertson** [comment]
Can you tell me where I can find the "js/sigma/sigma.parseJson.js" file to do this? I just used the Geolayout in Gephi and it i plotting upside down. Thanks!

**Comment by computermacgyver** [comment]
Hi @EROB777 this thread is about the sigma.js exporter and how the graph renders in a web browser. If the graph is upside down in Gephi, this thread is not applicable.

**Comment by gaardhus** [comment]
> 
> 
> @NoahGordon I was waiting on upstream, Sigma.js, and ideally wanted to see this fix there to avoid having different versions of sigma.js floating around. I consider this bug low priority as, in general, graph layout is invariant to the exact position of nodes (i.e., a flipped graph is isomorphically the same graph and the flip doesn't affect its presentation usually [geographic layouts being the one counter-example I've seen]).
> 
> Nonetheless, I may address this in an interm release while waiting on upstream. Probably the easiest fix is to alter the js/sigma/sigma.parseJson.js file to add the second line as follows:
> 
> ```
> var id=data.nodes[i].id;
> data.nodes[i].y=-data.nodes[i].y;
> ```

This was super useful! Would be awesome if this was implemented by default :)

**Comment by ocambra** [comment]
I have similar problems when exporting a GeoLayer graph with any of the web publishing plugins, it exports upside down, compared to my view in Gephi. I tried to change "js/sigma/sigma.parseJson.js" file to add the second line as @computermacgyver proposed but not results at all. Any idea how to fix this problem?

**Comment by computermacgyver** [comment]
Hi @ocambra , Let's troubleshoot on #80 . 

For anyone else finding this thread, the `js/sigma/sigma.parseJson.js` should look like [the version here](https://github.com/oxfordinternetinstitute/InteractiveVis/blob/master/network/js/sigma/sigma.parseJson.js#L10), which has the fix already in it. 

I'm also happy to say that the JSON export will be directly in Gephi very soon - https://github.com/gephi/gephi/issues/2674
